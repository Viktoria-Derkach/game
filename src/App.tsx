import React, { FC } from 'react';

import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
  Redirect,
} from 'react-router-dom';

import { MinesweeperWithHooks } from '@/pages/MinesweeperWithHooks';

export const App: FC = () => (
  <Router>
    <nav>
      <ul>
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="/game-with-hooks">Game With Hooks</Link>
        </li>
      </ul>
    </nav>
    <Switch>
      <Route exact path="/">
        <Home />
      </Route>
      <Route path="/game-with-hooks/:username?">
        <MinesweeperWithHooks />
      </Route>
      <Route path="*">
        <Redirect to="/" />
      </Route>
    </Switch>
  </Router>
);

const Home: FC = () => <h2>Minesweeper game Forever!</h2>;
