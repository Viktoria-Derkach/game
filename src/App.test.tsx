import React from 'react';
import { render } from '@testing-library/react';

import { App, Home } from './App';

describe('App test cases', () => {
  it('App renders check', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Home renders check', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
