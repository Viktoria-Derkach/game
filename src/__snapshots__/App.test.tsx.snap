// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test cases App renders check 1`] = `
<DocumentFragment>
  <nav>
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/game-with-hooks"
        >
          Game With Hooks
        </a>
      </li>
      <li>
        <a
          href="/game-with-usereducer"
        >
          Game With useReducer
        </a>
      </li>
      <li>
        <a
          href="/game-with-reactredux"
        >
          Game With ReactRedux
        </a>
      </li>
    </ul>
  </nav>
  <h2>
    Minesweeper game Forever!
  </h2>
</DocumentFragment>
`;

exports[`App test cases Home renders check 1`] = `
<DocumentFragment>
  <h2>
    Minesweeper game Forever!
  </h2>
</DocumentFragment>
`;
